test/byContract/IexecEscrow/IexecEscrowToken.test.ts

```diff
@@ -0,0 +1,473 @@
+        });
+        it('Should withdraw To with zero token', async () => {
+            const withdrawToParams = {
+                amount: 0,
+                target: accountB.address,
+            };
+            const withdrawToArgs = Object.values(withdrawToParams) as [BigNumber, string];
+
+            expect(await iexecPocoAsAccountA.callStatic.withdrawTo(...withdrawToArgs)).to.be.true;
+            await expect(iexecPocoAsAccountA.withdrawTo(...withdrawToArgs))
+                .to.changeTokenBalances(rlcInstance, [iexecPoco, accountB], [-0, 0])
+                .to.emit(rlcInstance, 'Transfer')
+                .withArgs(iexecPoco, withdrawToParams.target, 0)
+                .to.changeTokenBalances(iexecPoco, [accountA], [-0])
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(accountA.address, AddressZero, 0);
+        });
+        it('Should not withdraw To tokens with empty balance', async () => {
+            await expect(
+                iexecPocoAsAccountA.withdrawTo(standardAmount, accountB.address),
+            ).to.be.revertedWithoutReason();
+        });
+        it('Should not withdraw To tokens with insufficient balance', async () => {
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, standardAmount)
+                .then((tx) => tx.wait());
+            await iexecPocoAsAccountA.deposit(standardAmount).then((tx) => tx.wait());
+
+            await expect(
+                iexecPocoAsAccountA.withdrawTo(standardAmount.mul(2), accountB.address),
+            ).to.be.revertedWithoutReason();
+        });
+    });
+
+    describe('Recover', () => {
+        it('Should recover from balance deviation', async () => {
+            await rlcInstance.connect(iexecAdmin).transfer(proxyAddress, standardAmount); // Simulate deviation
+
+            const initTotalSupply = await iexecPoco.totalSupply();
+            const expectedDelta = standardAmount;
+
+            await expect(iexecPocoAsAdmin.recover())
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(AddressZero, iexecAdmin.address, expectedDelta);
+
+            expect(await iexecPoco.totalSupply()).to.equal(initTotalSupply.add(expectedDelta));
+        });
+        it('Should recover 0 token when balance matches total supply', async () => {
+            const initialSupply = await iexecPoco.totalSupply();
+
+            await expect(iexecPocoAsAdmin.recover())
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(AddressZero, iexecAdmin.address, 0);
+            expect(await iexecPoco.totalSupply()).to.equal(initialSupply);
+        });
+        it('Should not allow non-owner to recover', async () => {
+            await expect(iexecPocoAsAccountA.recover()).to.be.revertedWith(
+                'Ownable: caller is not the owner',
+            );
+        });
+    });
+
+    describe('receiveApproval', () => {
+        it('Should receiveApproval', async () => {
```

Comment:
I might suggest to use that instead: https://github.com/iExecBlockchainComputing/rlc-token/blob/master/contracts/RLC.sol#L84

Reply:
Mmmh the function `approveAndCall` from `IexecERC20Delegate.sol` is tested in the [execERC20 UT](https://github.com/iExecBlockchainComputing/PoCo/blob/7b748a63bf6b32f7c28e336d3327660060b993a3/test/byContract/IexecERC20/IexecERC20.test.ts#L117)